plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts

```diff
@@ -569,6 +561,66 @@ export function addHistoricalEventsExportCapabilityV2(
         }
     }
 
+    async function handleFetchError(
```

Comment:
I'm not a fan of this being almost a copy of `handleExportError` could we unify?

Reply:
I'd rather keep separate for now. There are enough annoying little changes that it becomes a web of `if`s if we try to keep consolidated.

Can you see a nice way to refactor that keeps the little variations e.g. in all the messages and metrics.